compiler/rustc_codegen_gcc/src/context.rs

```diff
@@ -477,7 +478,10 @@ impl<'gcc, 'tcx> LayoutOfHelpers<'tcx> for CodegenCx<'gcc, 'tcx> {
     #[inline]
     fn handle_layout_err(&self, err: LayoutError<'tcx>, span: Span, ty: Ty<'tcx>) -> ! {
         if let LayoutError::SizeOverflow(_) = err {
-            self.sess().span_fatal(span, &err.to_string())
+            let _ = respan(span, err);
+            // error[E0277]: the trait bound `Spanned<rustc_middle::ty::layout::LayoutError<'tcx>>: SessionDiagnostic<'_, !>` is not satisfied
```

Comment:
@davidtwco I'm honestly not sure if I'm on the right track, but I tried to implement `SessionDiagnostic` for `LayoutError` and `Spanned<T>` and I'm getting this error. Is this what you were thinking?